File: kirun-js/src/engine/json/schema/type/SingleType.ts
import { SchemaType } from './SchemaType';
import { Type } from './Type';

export class SingleType extends Type {
    private type: SchemaType;

    public constructor(type: SchemaType | SingleType) {
        super();

        if (type instanceof SingleType) this.type = (type as SingleType).type;
        else this.type = type as SchemaType;
    }

    public getType(): SchemaType {
        return this.type;
    }

    public getAllowedSchemaTypes(): Set<SchemaType> {
        return new Set([this.type]);
    }

    public contains(type: SchemaType): boolean {
        return this.type == type;
    }
}

File: kirun-py/src/kirun_py/json/schema/type/single_type.py
from __future__ import annotations
from typing import Set, Union

from kirun_py.json.schema.type.schema_type import SchemaType
from kirun_py.json.schema.type.type_ import Type


class SingleType(Type):
    def __init__(self, type_: Union[SchemaType, SingleType]):
        if isinstance(type_, SingleType):
            self._type = type_._type
        else:
            self._type = type_

    def get_type(self) -> SchemaType:
        return self._type

    def get_allowed_schema_types(self) -> Set[SchemaType]:
        return {self._type}

    def contains(self, type_: SchemaType) -> bool:
        return self._type == type_
